eodatasets3/images.py

```diff
@@ -686,6 +683,64 @@ def create_thumbnail(
                                 index,
                             )
 
+    def create_thumbnail_singleband(
+        self,
+        in_file: Path,
+        out_file: Path,
+        bit: int = None,
+        lookup_table: Dict[int, Tuple[int, int, int]] = None,
+    ):
+        """
+        Write out a JPG thumbnail from a singleband image.
+        This takes in a path to a valid raster dataset and writes
+        out a file with only the values of the bit (integer) as white
+        """
+        assert not (
+            bit is not None and lookup_table is not None
+        ), "Please set either bit or lookup_table, and not both of them"
+
+        with rasterio.open(in_file) as dataset:
+            data = dataset.read()
+
+            if bit is not None:
+                out_data = numpy.copy(data)
+                out_data[data != bit] = 0
+                stretch = [0, bit]
+            if lookup_table is not None:
+                out_data = [
+                    numpy.full_like(data, 0),
+                    numpy.full_like(data, 0),
+                    numpy.full_like(data, 0),
+                ]
+                stretch = [0, 255]
+
+                for value, rgb in lookup_table.items():
+                    for index in range(3):
+                        out_data[index][data == value] = rgb[index]
+
+        meta = dataset.meta
+        meta["driver"] = "GTiff"
+
+        if bit:
+            interim_file = Path(tempfile.gettempdir()) / "temp.tif"
```

Comment:
I missed this before -- but this will use the same temp file if two processes run concurrently.

We use the `TemporaryDirectory` elsewhere to avoid this (it also cleans up automatically), eg:

https://github.com/GeoscienceAustralia/eo-datasets/blob/cbc21162134214b701c3ffb60e0e4ee8feb42cc7/eodatasets3/images.py#L552-L554

(Those parameters might be optional in this case. Setting the output folder was to avoid errors of limited `/tmp` space when running huge parallel jobs on a machine.)

Reply:
Also good point.